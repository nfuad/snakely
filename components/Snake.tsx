import React from 'react'

export default (props) => {
  return (
    <div>
      {props.snakeDots.map((dot, i) => {
        const style = {
          left: `${dot[0]}%`,
          top: `${dot[1]}%`
        }
        return (
          <div className="snake-dot" key={i} style={style}>
            <style jsx global>{`
              .snake-dot {
                position: absolute;
                width: 2%;
                height: 2%;
                background-color: #000;
                border: 1px solid #fff;
                z-index: 2;
              }
            `}</style>
          </div>
        )
      })}
    </div>
  )
}
