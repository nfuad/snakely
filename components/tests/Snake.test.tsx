import * as React from 'react'
import { shallow, mount, render } from 'enzyme'

import Snake from '../Snake'

const mockData = {
  snakeDots: [
    [0, 0],
    [2, 0]
  ]
}

describe('Snake component', () => {
  it('renders', () => {
    shallow(<Snake snakeDots={mockData.snakeDots} />)
  })

  it('matches the snapshot', () => {
    const component = mount(<Snake snakeDots={mockData.snakeDots} />)
    expect(component).toMatchSnapshot()
  })

  it('updates via props', () => {
    const component = mount(<Snake snakeDots={mockData.snakeDots} />)
    expect(component).toMatchSnapshot()

    component.setProps({
      snakeDots: [
        [0, 2],
        [2, 2]
      ]
    })

    expect(component).toMatchSnapshot()
  })
})
