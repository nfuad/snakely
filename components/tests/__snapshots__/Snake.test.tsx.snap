// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snake component matches the snapshot 1`] = `
<_default
  snakeDots={
    Array [
      Array [
        0,
        0,
      ],
      Array [
        2,
        0,
      ],
    ]
  }
>
  <div>
    <div
      className="jsx-1162698439 snake-dot"
      key="0"
      style={
        Object {
          "left": "0%",
          "top": "0%",
        }
      }
    >
      <JSXStyle
        id="1162698439"
      />
    </div>
    <div
      className="jsx-1162698439 snake-dot"
      key="1"
      style={
        Object {
          "left": "2%",
          "top": "0%",
        }
      }
    >
      <JSXStyle
        id="1162698439"
      />
    </div>
  </div>
</_default>
`;

exports[`Snake component updates via props 1`] = `
<_default
  snakeDots={
    Array [
      Array [
        0,
        0,
      ],
      Array [
        2,
        0,
      ],
    ]
  }
>
  <div>
    <div
      className="jsx-1162698439 snake-dot"
      key="0"
      style={
        Object {
          "left": "0%",
          "top": "0%",
        }
      }
    >
      <JSXStyle
        id="1162698439"
      />
    </div>
    <div
      className="jsx-1162698439 snake-dot"
      key="1"
      style={
        Object {
          "left": "2%",
          "top": "0%",
        }
      }
    >
      <JSXStyle
        id="1162698439"
      />
    </div>
  </div>
</_default>
`;

exports[`Snake component updates via props 2`] = `
<_default
  snakeDots={
    Array [
      Array [
        0,
        2,
      ],
      Array [
        2,
        2,
      ],
    ]
  }
>
  <div>
    <div
      className="jsx-1162698439 snake-dot"
      key="0"
      style={
        Object {
          "left": "0%",
          "top": "2%",
        }
      }
    >
      <JSXStyle
        id="1162698439"
      />
    </div>
    <div
      className="jsx-1162698439 snake-dot"
      key="1"
      style={
        Object {
          "left": "2%",
          "top": "2%",
        }
      }
    >
      <JSXStyle
        id="1162698439"
      />
    </div>
  </div>
</_default>
`;
